src/sentry/taskworker/registry.py

```diff
@@ -0,0 +1,159 @@
+    All tasks within a namespace are stored in the same topic and run by shared
+    worker pool.
+    """
+
+    __registered_tasks: dict[str, Task]
+    __producer: KafkaProducer | None = None
+
+    def __init__(self, name: str, topic: str, deadletter_topic: str, retry: Retry | None):
+        # TODO(taskworker) implement default deadlines for tasks
+        self.name = name
+        self.topic = topic
+        self.deadletter_topic = deadletter_topic
+        self.default_retry = retry
+        self.__registered_tasks = {}
+
+    @property
+    def producer(self) -> KafkaProducer:
+        if self.__producer:
+            return self.__producer
+        cluster_name = get_topic_definition(Topic.TASK_WORKER)["cluster"]
+        producer_config = get_kafka_producer_cluster_options(cluster_name)
+        self.__producer = KafkaProducer(producer_config)
+
+        return self.__producer
+
+    @producer.setter
+    def producer(self, producer: KafkaProducer) -> None:
+        self.__producer = producer
+
+    @producer.deleter
+    def producer(self) -> None:
+        self.__producer = None
+
+    def get(self, name: str) -> Task:
+        if name not in self.__registered_tasks:
+            raise KeyError(f"No task registered with the name {name}. Check your imports")
+        return self.__registered_tasks[name]
+
+    def contains(self, name: str) -> bool:
+        return name in self.__registered_tasks
+
+    def register(
+        self,
+        name: str,
+        idempotent: bool | None = None,
+        deadline: timedelta | int | None = None,
+        retry: Retry | None = None,
+    ):
+        """register a task, used as a decorator"""
+
+        def wrapped(func):
+            task = Task(
+                name=name,
+                func=func,
+                namespace=self,
+                idempotent=idempotent,
+                deadline=deadline,
+                retry=retry or self.default_retry,
+            )
+            # TODO(taskworker) tasks should be registered into the registry
+            # so that we can ensure task names are globally unique
+            self.__registered_tasks[name] = task
+            return task
+
+        return wrapped
+
+    def retry_task(self, taskdata: TaskActivation) -> None:
+        self.producer.produce(
+            ArroyoTopic(name=self.topic),
+            KafkaPayload(key=None, value=taskdata.SerializeToString(), headers=[]),
+        )
+
+    def send_task(self, task: Task, args, kwargs) -> None:
+        task_message = self._serialize_task_call(task, args, kwargs)
+        # TODO(taskworker) this could use an RPC instead of appending to the topic directly
+        # TODO(taskworker) callback handling
+        self.producer.produce(
+            ArroyoTopic(name=self.topic),
+            KafkaPayload(key=None, value=task_message, headers=[]),
+        )
+
+    def _serialize_task_call(self, task: Task, args: list[Any], kwargs: Mapping[Any, Any]) -> bytes:
```

Comment:
`list[Any]`, `Mapping[Any, Any]` makes me nervous here -- can we find a way to make this strongly typed? maybe genericized based on the `Task` ?

Reply:
I've moved the task call serialization onto `Task`. This removed a bunch of Any's and helped contain types better.